src/torrent.cpp

```diff
@@ -7429,23 +7429,51 @@ bool is_downloading_state(int const st)
 		if (lhs->on_parole() != rhs->on_parole())
 			return lhs->on_parole();
 
-		// prefer to disconnect peers that send data at a lower rate
+		// prefer to disconnect peers that chokes us
+		if (lhs->is_choked() != rhs->is_choked())
+			return lhs->is_choked();
```

Comment:
this is a bit risky. a peer can unchoke you but still not send you any data. A peer that has temporarily choked you, but otherwise is a good peer, would be penalized

Reply:
In the existing code, it is unlikely that choked peers are ever disconnected because the previous check is on speed, and it is unlikely that the speed of two connections is the same esp. if one is choked.

However, I can see that a temporary choke on a fast connection might be preferable to a slower connection which is not choked.

I can either move this code back to where it was or drop it.

Reply:
I think it would make sense to drop this condition, if the aspiration is to come up with something better and less crude. In general it's probably a good idea to primarily look at actual useful data received, and no second or third order indicators.

Reply:
Ok - will fix this tomorrow.